src/sensors/src/ThreeAxisForceTorqueContactSensor.cpp

```diff
@@ -0,0 +1,232 @@
+        return false;
+    }
+
+    return true;
+}
+
+Sensor* ThreeAxisForceTorqueContactSensor::clone() const
+{
+    return (Sensor *)new ThreeAxisForceTorqueContactSensor(*this);
+}
+
+bool ThreeAxisForceTorqueContactSensor::updateIndices(const Model& model)
+{
+    iDynTree::LinkIndex linkNewIndex = model.getLinkIndex(this->pimpl->parent_link_name);
+
+    if( linkNewIndex == iDynTree::LINK_INVALID_INDEX )
+    {
+        return false;
+    }
+
+    this->pimpl->parent_link_index = linkNewIndex;
+
+    return true;
+}
+
+std::string ThreeAxisForceTorqueContactSensor::getName() const
+{
+    return this->pimpl->name;
+}
+
+SensorType ThreeAxisForceTorqueContactSensor::getSensorType() const
+{
+    return THREE_AXIS_FORCE_TORQUE_CONTACT;
+}
+
+Transform ThreeAxisForceTorqueContactSensor::getLinkSensorTransform() const
+{
+    return(this->pimpl->link_H_sensor);
+}
+
+void ThreeAxisForceTorqueContactSensor::setLoadCellLocations(std::vector<Position> &loadCellLocations)
+{
+    pimpl->m_loadCellLocations = loadCellLocations;
+}
+
+std::vector<Position>  ThreeAxisForceTorqueContactSensor::getLoadCellLocations()
+{
+    return pimpl->m_loadCellLocations;
+}
+
+Vector3  ThreeAxisForceTorqueContactSensor::computeThreeAxisForceTorqueFromLoadCellMeasurements(VectorDynSize& loadCellMeasurements)
+{
+    if (loadCellMeasurements.size() != pimpl->m_loadCellLocations.size())
+    {
+        reportError("ThreeAxisForceTorqueContactSensor","computeThreeAxisForceTorqueFromLoadCellMeasurements", "loadCellMeasurements has the wrong size");
+        Vector3 zero; zero.zero();
+        return zero;
+    }
+
+    Wrench totalWrench;
+    totalWrench.zero();
+
+    for (size_t i=0; i < loadCellMeasurements.size(); i++)
+    {
+        // We assume that the total forcetorque is given by the sum of pure forces on the z axis measured by the load cells
+        Wrench loadCellWrench = Wrench(LinearForceVector3(0.0, 0.0, loadCellMeasurements(i)*1.0), AngularForceVector3(0.0, 0.0, 0.0));
+        Transform sensor_H_loadCell(Rotation::Identity(), pimpl->m_loadCellLocations[i]);
+        totalWrench = totalWrench + sensor_H_loadCell*loadCellWrench;
+    }
+
+    // Project the force torque
+    Vector3 ret;
+    ret(0) = totalWrench.getLinearVec3()(2);
+    ret(1) = totalWrench.getAngularVec3()(0);
+    ret(2) = totalWrench.getAngularVec3()(1);
+
+    return ret;
+}
+
+Position ThreeAxisForceTorqueContactSensor::computeCenterOfPressureFromLoadCellMeasurements(VectorDynSize& loadCellMeasurements)
+{
+    if (loadCellMeasurements.size() != pimpl->m_loadCellLocations.size())
+    {
+        reportError("ThreeAxisForceTorqueContactSensor","computeThreeAxisForceTorqueFromLoadCellMeasurements", "loadCellMeasurements has the wrong size");
+        Position zeroVec; zeroVec.zero();
+        return zeroVec;
+    }
+
+    // We assume that the location of the load cells is on the XY plane. In this case, the cop is just the average of the locations weighted on the measurements
```

Comment:
Just a question: this should also match to the result computed by `computeThreeAxisForceTorqueFromLoadCellMeasurements` and properly manipulated, right? (like torque / f_z)
Can it be used as a consistency test?

Reply:
Yes, indeed.